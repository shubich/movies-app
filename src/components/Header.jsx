import React from 'react';
import { Route, Switch } from 'react-router-dom';
import {Nav} from './Nav';
import {SearchForm} from './SearchForm';
import {FilmDetails} from './FilmDetails';

export const Header = () => (
    <header id='header'>
        <div className='container'>
            <Nav />
            <Switch>
                <Route exact path='/' component={SearchForm}/>
                <Route exact path='/search' component={SearchForm}/>
                <Route path='/movie/:id' component={FilmDetails}/>
            </Switch>
        </div>
    </header>
);