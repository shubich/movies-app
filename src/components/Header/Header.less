@import "../../styles/palette";
@import "../../styles/spaces";
@import "../../styles/media";

.Header {
    background: @gray;
    color: white;
    padding: @desktop-space;
    @media @phone {
        padding: @phone-space;
    }   
}