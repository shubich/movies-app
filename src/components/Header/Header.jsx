import React from 'react';
import PropTypes from 'prop-types';

import './Header.less';

const Header = props => (
  <header className="Header">
    <div className="Header-MaskLayer">
      <div className="Container">
        {props.children}
      </div>
    </div>
  </header>
);

Header.propTypes = {
  children: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Header;
