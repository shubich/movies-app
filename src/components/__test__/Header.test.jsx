import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

let props;

describe('Header', () => {
  beforeEach(() => {
    props = {
      children: 'test',
    };
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
