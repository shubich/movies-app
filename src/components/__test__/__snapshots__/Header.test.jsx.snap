// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header>
    test
  </Header>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        test
      </div>,
      "id": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "test",
        "className": "container",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          test
        </div>,
        "id": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "container",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
